File: ts/1.array/tandem-bicycle.ts
function compare(max: boolean, a: number, b: number) {
    if (max) {
        return a > b ? a : b;
    }
    return a > b ? b : a;
}

export function tandemBicycle(
    rs: number[],
    bs: number[],
    max: boolean
): number {
    rs.sort((a, b) => (a > b ? 1 : -1));
    bs.sort((a, b) => (a > b ? 1 : -1));
    let speed = 0;

    for (let i = 0; i < rs.length; ++i) { 
        const bv = bs[bs.length - 1 - i];
        const rv = rs[i];

        speed += compare(max, rv, bv);
    }

    return speed;
}

File: golang/1.array/tandem_bicycle.go
package main

import "sort"

func compare(mx bool, a,b int) int {
    if mx { 
        if a > b { 
            return a 
        }

        return b
    }

    if a > b { 
        return b
    }

    return a
}

func TandemBicycle(rs, bs []int, mx bool) int {
    sort.Ints(rs)
    sort.Ints(bs)
    S := 0


    for i:=0; i < len(rs) ; i++ { 
        rsv := rs[i] 
        bsv := bs[len(bs) -1 - i]
        
        S += compare(mx, rsv, bsv)

    }

    return S 
}
